fund.go

```diff
@@ -0,0 +1,52 @@
+package fineractor
+
+import (
+	"log"
+)
+
+type FundIncrementRequest struct {
+	Locale            string `json:"locale"`
+	DateFormat        string `json:"dateFormat"`
+	TransactionDate   string `json:"transactionDate"`
+	TransactionAmount string `json:"transactionAmount"`
+	PaymentTypeId     string `json:"paymentTypeId"`
+}
+
+type FundIncrementResponse struct{}
+
+type FundDecrementRequest struct{}
+
+type FundDecrementResponse struct{}
+
+type FundValueRequest struct{}
+
+type FundValueResponse struct{}
+
+type FundAvailablityRequest struct{}
+
+type FundAvailablityResponse struct{}
+
+type FundsRequest struct{}
+
+type FundsResponse struct{}
+
+func (client Client) FundIncrement(request FundIncrementRequest) (FundIncrementResponse, error) {
```

Comment:
Pointer receivers please.

Reply:
@meson10 It cannot be a pointer receiver as NewClient or NewMockClient does not return a pointer 
It returns a interface which cannot be of type pointer 
Further, we are not modifying the Client or MockClient object in anyway, so passing it as pointer is not required.

Reply:
- It does not matter whether you return an interface. It can still be a pointer receiver.

- Why it might matter here? Because you would reuse an HTTP Client when you have to make an actual call and not make a new Connection every time. Infact NewClient should use sync.Once to only initialize a client once and keep returning that which internally uses a shared http.Client.

- Alternate implementation is to have a module level Client variable and access that from the struct which can have Value receivers.

- In general, this might be an informative read: https://golang.org/doc/faq#methods_on_values_or_pointers